pkg/resources/rbac/clusterrole_test.go

```diff
@@ -0,0 +1,151 @@
+			Name: "my-cluster-role",
+		},
+	}
+
+	t.Run("Create new ClusterRole", func(t *testing.T) {
+		// fake client
+		client := fake.NewClientBuilder().WithScheme(scheme.Scheme).Build()
+
+		err := SyncClusterRole(ctx, clusterRole, client)
+		assert.NoError(t, err)
+
+		// check that the cluster role was created
+		foundClusterRole := &rbacv1.ClusterRole{}
+		err = client.Get(ctx, types.NamespacedName{Name: clusterRole.Name, Namespace: clusterRole.Namespace}, foundClusterRole)
+		assert.NoError(t, err)
+	})
+
+	t.Run("Handle error on getting clusterRole", func(t *testing.T) {
+		client := &MockClient{
+			GetFunc: func(_ context.Context, _ client.ObjectKey, _ client.Object, _ ...client.GetOption) error {
+				return errors.New("get error")
+			},
+		}
+
+		err := SyncClusterRole(ctx, clusterRole, client)
+		assert.Error(t, err)
+		assert.Equal(t, "get error", err.Error())
+	})
+
+	t.Run("Handle error on creating clusterRole", func(t *testing.T) {
+		client := &MockClient{
+			GetFunc: func(_ context.Context, key client.ObjectKey, _ client.Object, _ ...client.GetOption) error {
+				return apierrors.NewNotFound(rbacv1.Resource("clusterrole"), key.Name)
+			},
+			CreateFunc: func(_ context.Context, _ client.Object, _ ...client.CreateOption) error {
+				return errors.New("create error")
+			},
+		}
+
+		err := SyncClusterRole(ctx, clusterRole, client)
+		assert.Error(t, err)
+		assert.Equal(t, "create error", err.Error())
+	})
+
+	t.Run("Handle error on getting creating clusterRole", func(t *testing.T) {
+		client := &MockClient{
+			CreateFunc: func(_ context.Context, _ client.Object, _ ...client.CreateOption) error {
+				return nil
+			},
+			GetFunc: func(_ context.Context, key client.ObjectKey, _ client.Object, _ ...client.GetOption) error {
+				return apierrors.NewNotFound(rbacv1.Resource("clusterrole"), key.Name)
+			},
+		}
+
+		err := SyncClusterRole(ctx, clusterRole, client)
+		assert.Error(t, err)
+		assert.Equal(t, "clusterrole.rbac.authorization.k8s.io \"my-cluster-role\" not found", err.Error())
+	})
+
+	t.Run("Handle existing clusterRole", func(t *testing.T) {
+		client := &MockClient{
+			GetFunc: func(_ context.Context, _ client.ObjectKey, _ client.Object, _ ...client.GetOption) error {
+				return nil
+			},
+		}
+
+		err := SyncClusterRole(ctx, clusterRole, client)
+		assert.NoError(t, err)
+	})
+
+	t.Run("Handle existing clusterRole update error", func(t *testing.T) {
+		client := &MockClient{
+			GetFunc: func(_ context.Context, _ client.ObjectKey, _ client.Object, _ ...client.GetOption) error {
+				return nil
+			},
+			UpdateFunc: func(_ context.Context, _ client.Object, _ ...client.UpdateOption) error {
+				return errors.New("update error")
+			},
+		}
+
+		err := SyncClusterRole(ctx, clusterRole, client)
+		assert.Error(t, err)
+		assert.Equal(t, "update error", err.Error())
+	})
+}
+
+// MockClient is a mock implementation of the client.Client interface for testing purposes.
+type MockClient struct {
```

Comment:
Should types like this be brought to the top of the file?

Reply:
Added to the top of common.go